pkg/webhooks/machine_webhook_test.go

```diff
@@ -3209,6 +3210,13 @@ func TestValidateGCPProviderSpec(t *testing.T) {
 			},
 			expectedOk: true,
 		},
+		{
+			testCase:         "with unknown fields in the providerSpec",
+			overrideRawBytes: []byte(`{"kind":"GCPMachineProviderSpec","apiVersion":"gcpprovider.openshift.io/v1beta1","metadata":{"creationTimestamp":null},"userDataSecret":{"name":"name"},"credentialsSecret":{"name":"name"},"canIPForward":false,"deletionProtection":false,"disks":[{"autoDelete":false,"boot":false,"sizeGb":16,"type":"","image":"","labels":null}],"networkInterfaces":[{"network":"network","subnetwork":"subnetwork"}],"serviceAccounts":[{"email":"email","scopes":["scope"]}],"machineType":"machineType","region":"region","zone":"region-zone","projectID":"projectID","gpus":[{"count":0,"type":"type"}],"onHostMaintenance":"Terminate","randomField-1": "something","randomField-2": "something"}`),
+			expectedOk:       true,
+			expectedError:    "",
+			expectedWarnings: []string{"providerSpec.value: Unsupported value: \"randomField-1\": Unknown field (randomField-1) will be ignored"},
```

Comment:
So in the warning we will return the first unknown field, correct? If there are multiple of those we won't alert on the others, is this by design or is it because `yaml.DisallowUnknownFields` doesn't return them all?

Reply:
The latter, I think returning all the fields is better but the error we get just return the first unknown field
see [here](https://go.dev/play/p/rl7VAMwBRho)

Reply:
Gotcha. Thanks for letting me know.